agent/agent.go

```diff
@@ -100,6 +100,11 @@ func NewAgentWithContext(ctx context.Context, config Config, logger hclog.Logger
 		redacts = redact.Flatten(hclRedacts, redacts)
 	}
 
+	// Ensure temporary directory exists
+	if _, err = os.Stat(config.TmpDir); err != nil {
+		return nil, fmt.Errorf("Error creating tmpdir for new agent: %w", err)
```

Comment:
Should the error say that the temporary directory doesn't exist, rather than saying it failed to create? It looks like we aren't creating it here, so the error might be misleading.